File: frontend/src/api/verification.ts
import { apiClient } from "./client";
import type { VerificationRequest, VerificationResponse } from "../types/api";

export async function verifyEligibility(
  payload: VerificationRequest,
): Promise<VerificationResponse> {
  const response = await apiClient.post<VerificationResponse>(
    "/verification/verify",
    payload,
  );

  return response.data;
}

File: backend/app/api/routes/verification.py
from fastapi import APIRouter

from app.schemas.verification import VerificationRequest, VerificationResponse
from app.services.verification_service import verify_eligibility

router = APIRouter()


@router.post("/verify", response_model=VerificationResponse)
def verify_eligibility_route(payload: VerificationRequest) -> VerificationResponse:
  return verify_eligibility(payload)
